scala/scala.bzl

```diff
@@ -249,56 +257,56 @@ scala_repl = rule(
     toolchains = ['@io_bazel_rules_scala//scala:toolchain_type'],
 )
 
-_SCALA_BUILD_FILE = """
-# scala.BUILD
-java_import(
-    name = "scala-xml",
-    jars = ["lib/scala-xml_2.11-1.0.5.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-parser-combinators",
-    jars = ["lib/scala-parser-combinators_2.11-1.0.4.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-library",
-    jars = ["lib/scala-library.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-compiler",
-    jars = ["lib/scala-compiler.jar"],
-    visibility = ["//visibility:public"],
-)
-
-java_import(
-    name = "scala-reflect",
-    jars = ["lib/scala-reflect.jar"],
-    visibility = ["//visibility:public"],
-)
-"""
-
 def scala_repositories():
-  native.new_http_archive(
-      name = "scala",
-      strip_prefix = "scala-2.11.12",
-      sha256 =
-      "b11d7d33699ca4f60bc3b2b6858fd953e3de2b8522c943f4cda4b674316196a8",
-      url = "http://downloads.lightbend.com/scala/2.11.12/scala-2.11.12.tgz",
-      build_file_content = _SCALA_BUILD_FILE,
+  _scala_maven_import_external(
+      name = "scala_library",
+      artifact = "org.scala-lang:scala-library:2.11.12",
+      jar_sha256 =
+      "0b3d6fd42958ee98715ba2ec5fe221f4ca1e694d7c981b0ae0cd68e97baf6dce",
+      licenses = ["notice"],
+      server_urls = ["http://central.maven.org/maven2"],
   )
-
-  # scalatest has macros, note http_jar is invoking ijar
-  native.http_jar(
+  _scala_maven_import_external(
+      name = "scala_compiler",
+      artifact = "org.scala-lang:scala-compiler:2.11.12",
+      jar_sha256 =
+      "3e892546b72ab547cb77de4d840bcfd05c853e73390fed7370a8f19acb0735a0",
+      licenses = ["notice"],
+      server_urls = ["http://central.maven.org/maven2"],
+  )
+  _scala_maven_import_external(
+      name = "scala_reflect",
+      artifact = "org.scala-lang:scala-reflect:2.11.12",
+      jar_sha256 =
+      "6ba385b450a6311a15c918cf8688b9af9327c6104f0ecbd35933cfcd3095fe04",
+      licenses = ["notice"],
+      server_urls = ["http://central.maven.org/maven2"],
+  )
+  _scala_maven_import_external(
       name = "scalatest",
-      url =
-      "https://mirror.bazel.build/oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/2.2.6/scalatest_2.11-2.2.6.jar",
-      sha256 =
+      artifact = "org.scalatest:scalatest_2.11:2.2.6",
+      jar_sha256 =
       "f198967436a5e7a69cfd182902adcfbcb9f2e41b349e1a5c8881a2407f615962",
+      licenses = ["notice"],
+      server_urls = ["http://central.maven.org/maven2"],
+  )
+
+  _scala_maven_import_external(
+      name = "scala_xml",
```

Comment:
These names are pretty short. Since they have to be globally unique what about io_bazel_rules_scala_scala_xml?

Reply:
Doesn’t this contradict canonical naming? Why do you think here we should preface with rules_scala?
Not sure I disagree btw, trying to gather my thoughts about it

Reply:
We don’t own the global namespace. So who is to say someone else didn’t point scala_xml at something else. If you prefix with io_bazel_rules_scala I think you are asking for it. I think we can argue that we do own that namespace. If we could solve the problem of a canonical maven encoding, this could be a solved problem.

Reply:
👍 on having longer names. Don't have strong opinions on whether it should be `io_bazel_rules_scala_scala_xml` or e.g. `org_scala_lang_modules_scala_xml`